test/test_nn.py

```diff
@@ -1799,29 +1799,27 @@ def test_parameterlistdict_setting_attributes(self):
         self.assertTrue(len(w) == 0)
 
     def test_parameterlistdict_pickle(self):
+        # warning from torch.load call in _load_from_bytes used in UntypedStorage.__reduce__
+        regex = re.escape("You are using `torch.load` with `weights_only=False`")
         m = nn.ParameterList(map(nn.Parameter, [torch.rand(2), torch.rand(2)]))
-        with warnings.catch_warnings(record=True) as w:
+        with self.assertWarnsRegex(FutureWarning, regex) as w:
```

Comment:
Nit (no need to capture the context if it's not used later
```suggestion
        with self.assertWarnsRegex(FutureWarning, WEIGHTS_ONLY_WARN):
```